hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java

```diff
@@ -285,6 +295,8 @@ private void addToBatchTransactionInfoWithTrace(String parentName,
   void flushTransactions() {
     while (isRunning.get() && canFlush()) {
       flushCurrentBuffer();
+      // Invalidate all the content in S3SecretManager cache.
+      s3SecretManager.clearS3Cache();
```

Comment:
We cannot just simply cleanup all entries at once, only the committed transactions need to be cleaned up. RocksDb table cache uses trxLogIndex as an epoch for the cache entry. 
Refer to cleanup cache method in this class. The idea is to track the committed trxs and only remove those from the cache.

Reply:
Thanks for the comment @sadanand48 

I believe that the cache entries are always cleared during each double buffer flush, which means that eventually, everything must be written to the database at some point in time, correct? Or is there a possibility that something might be left behind in the cache? 
Please correct me if I am going wrong somewhere!

Reply:
```java
      buffer.iterator().forEachRemaining(
          entry -> addCleanupEntry(entry, cleanupEpochs));
```
Let's say there were entries added to the buffer after this line, In that case we cannot cleanup all entries in cache as cache might contain entries that have not yet been considered for flush in the batch. To solve this we maintain cleanup epochs and only delete those particular entries for cleanup from cache that have been flushed.